tools/chafa/chafa.c

```diff
@@ -178,110 +183,96 @@ interruptible_usleep (gdouble us)
     }
 }
 
-#ifdef G_OS_WIN32
-
-/* We must determine if stdout is redirected to a file, and if so, use a
- * different set of I/O functions. */
-static gboolean win32_stdout_is_file = FALSE;
-
-static gboolean
-safe_WriteConsoleA (HANDLE chd, const gchar *data, gsize len)
-{
-    gsize total_written = 0;
-
-    if (chd == INVALID_HANDLE_VALUE)
-        return FALSE;
-
-    while (total_written < len)
-    {
-        DWORD n_written = 0;
-
-        if (win32_stdout_is_file)
-        {
-            /* WriteFile() and fwrite() seem to work equally well despite various
-             * claims that the former does poorly in a UTF-8 environment. The
-             * resulting files look good in my tests, but note that catting them
-             * out with 'type' introduces lots of artefacts. */
-#if 0
-            if (!WriteFile (chd, data, len - total_written, &n_written, NULL))
-                return FALSE;
-#else
-            if ((n_written = fwrite (data, 1, len - total_written, stdout)) < 1)
-                return FALSE;
-#endif
-        }
-        else
-        {
-            if (!WriteConsoleA (chd, data, len - total_written, &n_written, NULL))
-                return FALSE;
-        }
-
-        data += n_written;
-        total_written += n_written;
-    }
-
-    return TRUE;
-}
 
-#endif
```

Comment:
I guess if we're saving the UTF-16 stuff for later, all you need is `canvas_to_conhost` for output, so this block doesn't need to be moved, and you can drop `safe_WriteConsoleW` too (or `#if 0` it out so we can use it when UTF-16 becomes independently selectable).